### ISSUE TITLE
Error when farming with small plot size

### ISSUE THREAD

**Issue by jfrank-summit** [subspace/subspace-cli#47]
I am getting an error when attempting to farm after wiping an existing plot. Steps to reproduce:

~~1. I have an existing plot~~
~~2. run `subspace-cli wipe`~~
3. run `subspace-cli init` with 1MB plot size
4. run `subspace-cli farm`

I receive the following error:
![image](https://user-images.githubusercontent.com/37932802/202234608-dc6da77b-e348-43c3-8a9b-f83d39fea918.png)

Update:
I am getting this error even when I don't have an existing plot (i.e. I clear all related files to start from scratch).

UPDATE 2:
This appears to be driven by plot size, 1MB doesn't gives the above error, 10MB works.

**Comment by ozgunozerk** [comment]
Thank you for submitting this. 
This is clearly a bug, I'll handle and try to put better error messages. 
However, as we spoke earlier, this is not unhandled panic, this format is what I'm expecting from the error handling (via color_eyre crate). 

For the error itself, I cannot reproduce this on macOS. What I'm suspecting is, the program may be confusing itself because of Windows+WSL setting. 
@jfrank-summit can you please attach another SS for the `init` step's `default directory`? I'm curious about what does it show for the default directory.

**Comment by jfrank-summit** [comment]
Here is the screenshot of the `init` step. Also, the files get created correctly (`init.toml` and `summary.toml`).
<img width="507" alt="image" src="https://user-images.githubusercontent.com/37932802/202247796-92b22d38-a85c-45e1-af2b-3ba710b8e82a.png">

Also, I know this one wasn't a panic. Haven't seen one since mentioning it to you.

**Comment by jfrank-summit** [comment]
> Ok wonderful. Is the default path `/root/.local/share/subspace-cli/plots` is there? If not, is it partially there? I'll check the code tomorrow and try to debug it tomorrow with the help of your answer :)

It is there. Thanks!

**Comment by jfrank-summit** [comment]
I was using 1MB plot size, but when I switch to 10MB it works. Is there a minimum plot size?

**Comment by ozgunozerk** [comment]
Following the error `Single disk plot creation error: I/O error: No such file or directory` -> the error must be originating in here: https://github.com/subspace/subspace/blob/main/crates/subspace-farmer/src/single_disk_plot.rs#L434, will update if I find anything significant.

**Comment by jfrank-summit** [comment]
@achiurizo was able to reproduce it on a machine running WSL, as was I on a separate machine also running WSL.

**Comment by ozgunozerk** [comment]
I'll try on my windows, and then windows + WSL, will report here. 
This bug originates in monorepo, so I don't think it's related to CLI (the origin at least). We haven't seen this bug on monorepo, since it does not allow any plot below 1GB. I'll put a restriction for plot size as well in CLI.

But still want to investigate this issue further, so don't close it :) 
There maybe an uncaught bug in monorepo (just maybe).

**Comment by ozgunozerk** [comment]
I've figured out the responsible line in monorepo, and that is actually calling a very low-level function in `fs2` crate. In our case, when plot size is lower than 10MB, it tries to allocate 0 bytes, and that results in platform-specific weird behaviour (it doesn't make sense to preallocate 0 bytes, so weird behaviour is somewhat acceptable). 
In short, having the check for plot size being larger than 1GB is for sure a solution, and there is not some weird bug related to this issue.
In short, we can safely close this one :)

**Comment by jfrank-summit** [comment]
Thank you @ozgunozerk !
